File: OnlineStoreAdminPanel/OnlineStoreAdminPanel.Web/Modules/ServerTypes/OnlineStore/ProductColumns.ts
﻿import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductRow } from "./ProductRow";

export interface ProductColumns {
    Id: Column<ProductRow>;
    Name: Column<ProductRow>;
    Description: Column<ProductRow>;
    Price: Column<ProductRow>;
    CatName: Column<ProductRow>;
    Photo: Column<ProductRow>;
    CreatedAt: Column<ProductRow>;
    Quan: Column<ProductRow>;
}

export class ProductColumns extends ColumnsBase<ProductRow> {
    static readonly columnsKey = 'OnlineStore.Product';
    static readonly Fields = fieldsProxy<ProductColumns>();
}

File: OnlineStoreAdminPanel/OnlineStoreAdminPanel.Web/Modules/OnlineStore/Product/ProductColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace OnlineStoreAdminPanel.OnlineStore.Columns;

[ColumnsScript("OnlineStore.Product")]
[BasedOnRow(typeof(ProductRow), CheckNames = true)]
public class ProductColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public string CatName { get; set; }
    public string Photo { get; set; }
    public DateTime CreatedAt { get; set; }
    public int Quan { get; set; }
}